src/transaction/locator_sr.c

```diff
@@ -12791,6 +12791,9 @@ redistribute_partition_data (THREAD_ENTRY * thread_p, OID * class_oid, int no_oi
 		  goto exit;
 		}
 
+              /* No supplemental log for insert is appended due to DDL statement */
+              thread_p->no_supplemental_log = true;
```

Comment:
You have to turn this off after a locator function using a heap logical function.

Reply:
Ok, I will turn off it after the locator function